### PR TITLE
Repo maintenance / harmonization

### PR DESCRIPTION
**Pull request by pvdlg**
- Switch from `standard` to `eslint` + `eslint-config-standard` + [prettier](https://github.com/prettier/prettier)
- Use ava and nyc + refactor test to independent from each other
- Add a License file + badge
- Remove npm and yarn lockfile
- Use a more generic `.gitignore` that would work on Windows, Linux, Mac OS

I also configured `babel-register` and `babel-polyfill` so the CLI will work on any Node version >=4 while we can continue to develop with all the last JS features.
It just takes 2 lines of code in the CLI entry point, and transpile transparently on execution, for the current Node version ran by the user.

@gr2m can you add the repo to codecov? I'm not admin of the orga anymore it seems.

### CODE REVIEW

**Comment by gr2m** [comment]
enabled repository on codecov.io, sorry for that. @christophwitzko said he was uncomfortable having you as owner as it would give you full control over https://github.com/semantic-release/go-semantic-release, but he mentioned that he fixed the authentication settings, seems it was not enough to install integrations. If that's a problem for you we can setup a call with Chris and discuss :)

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+package-lock=false
```
Comment (gr2m) [inline .npmrc]: I wish we could get rid of this file. But I guess it is the only way that will prevent the package-lock.json from being generated?

Comment (pvdlg) [reply .npmrc]: Yes. Or at least the only I found.
The `package-lock.json` is also present in the `.gitignore` so it won't be pushed to the repo.

The `package-lock=false` prevent the `package-lock.json` to be generated locally. So when you do `npm install` on a repo that you cloned a while, you'll get the last dependencies.

If we were to remove it, maintainers/contributors would have to do `rm package-lock.json && rm node_modules && npm install` to get the last dependencies. Probably not a big deal.

With or without it, it's up to the maintainers/contributors to do update their local dependencies regularly. And I'm not sure their an automated way to make sure everyone have updated dependencies...

**Comment by pvdlg** [comment]
> enabled repository on codecov.io, sorry for that. @christophwitzko said he was uncomfortable having you as owner as it would give you full control over https://github.com/semantic-release/go-semantic-release, but he mentioned that he fixed the authentication settings, seems it was not enough to install integrations. If that's a problem for you we can setup a call with Chris and discuss :)

No problem. I'm just going to bother you for this kind of tasks ;)

**Comment by gr2m** [comment]
with node 4

```
Gregors-MacBook:semantic-release-cli-test gregor$ semantic-release-cli setup
? What is your npm registry? https://registry.npmjs.org/
? What is your npm username? gr2m
? What is your npm password? [hidden]
WARN semantic-release keytar is not installed correctly, not saving password
? What is your GitHub username? gr2m
? What is your GitHub password? [hidden]
? What is your GitHub two-factor authentication code? 303445
WARN semantic-release keytar is not installed correctly, not saving password
? What CI are you using? Travis CI
ERR! semantic-release Error: Could not get repo id
ERR! semantic-release     at _callee5$ (/Users/gregor/Projects/semantic-release-cli/src/lib/travis.js:109:29)
ERR! semantic-release     at tryCatch (/Users/gregor/Projects/semantic-release-cli/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:65:40)
ERR! semantic-release     at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/gregor/Projects/semantic-release-cli/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:303:22)
ERR! semantic-release     at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/gregor/Projects/semantic-release-cli/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:117:21)
ERR! semantic-release     at step (/Users/gregor/Projects/semantic-release-cli/src/lib/travis.js:279:191)
ERR! semantic-release     at /Users/gregor/Projects/semantic-release-cli/src/lib/travis.js:279:361
ERR! semantic-release     at run (/Users/gregor/Projects/semantic-release-cli/node_modules/core-js/modules/es6.promise.js:66:22)
ERR! semantic-release     at /Users/gregor/Projects/semantic-release-cli/node_modules/core-js/modules/es6.promise.js:79:30
ERR! semantic-release     at flush (/Users/gregor/Projects/semantic-release-cli/node_modules/core-js/modules/_microtask.js:18:9)
ERR! semantic-release     at nextTickCallbackWith0Args (node.js:489:9)
ERR! semantic-release     at process._tickCallback (node.js:418:13)
ERR! semantic-release  { [Error: Could not get repo id]
ERR! semantic-release   stack: 'Error: Could not get repo id\n    at _callee5$ (/Users/gregor/Projects/semantic-release-cli/src/lib/travis.js:109:29)\n    at tryCatch (/Users/gregor/Projects/semantic-release-cli/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:65:40)\n    at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/gregor/Projects/semantic-release-cli/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:303:22)\n    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/gregor/Projects/semantic-release-cli/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:117:21)\n    at step (/Users/gregor/Projects/semantic-release-cli/src/lib/travis.js:279:191)\n    at /Users/gregor/Projects/semantic-release-cli/src/lib/travis.js:279:361\n    at run (/Users/gregor/Projects/semantic-release-cli/node_modules/core-js/modules/es6.promise.js:66:22)\n    at /Users/gregor/Projects/semantic-release-cli/node_modules/core-js/modules/es6.promise.js:79:30\n    at flush (/Users/gregor/Projects/semantic-release-cli/node_modules/core-js/modules/_microtask.js:18:9)\n    at nextTickCallbackWith0Args (node.js:489:9)\n    at process._tickCallback (node.js:418:13)' }
Gregors-MacBook:semantic-release-cli-test gregor$ node -v
v4.8.5
Gregors-MacBook:semantic-release-cli-test gregor$ semantic-release-cli --version
0.0.0-development
```

It worked well with latest Node 8. After 8 I tested it with Node 6 (on same test repository: https://github.com/gr2m/semantic-release-cli-test) and it worked without the error

I can help investigating the error but will have a look at the other PRs / issues first

**Comment by pvdlg** [comment]
This bug seems a big random. When it happens, it happens only after creating a repo on Github, cloning it locally, and running the command. Every subsequent try works.
I tested the following scenario:
- Node 4 => Happen most of the time
- Node 6 => Happen most of the time
- Node 8 => Happen but a bit less often
- Node 8 without `babel-register` and `babel-polyfill` => Happen but a bit less than on Node 4 and 6
- Code from `master` on Node 4 => Happen as well
- Code from `master` on Node 8 => Happen as well
- Current version installed with `npm -g semantic-release-cli` First try didn't happen. Happened on second try.

My estimate of how often it happens is really approximative, as I tested 2-3 times each scenario.

When I add logs this way:
```js
  const response = await promisify(travis.repos(info.ghrepo.slug[0], info.ghrepo.slug[1]).get.bind(travis))();
  travis.repoid = _.get(response, 'repo.id');
  console.log(response);
```

I obtain this in the console:
binary
When I retry it logs a json response as expected.

I'm fairly confident this bug is not a regression due to this PR.

**Comment by pvdlg** [comment]
Opened a bug for that: #147

**Pull request merged**